src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+
+### Graphics in Style Dictionary
+
+This is an interesting topic that deserves its own article. I will briefly touch on it here as it relates to dark mode. Design tokens can encapsulate things more complex than a color or size, they could be a whole graphic or icon too. For example you could create a logo design token that points to an SVG file:
+
+```json
+{
+  "image": {
+    "logo": { "value": "assets/svg/logo.svg" }
+  }
+}
+```
+
+That SVG file could then reference other design tokens:
+
+```xml
+<!-- assets/svg/logo.svg -->
+<svg width="100" height="100" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
+  <path
+    d="M5.03278 19C5.54736 11.182 12.0517 5 20 5C27.9482 5 34.4526 11.182 34.9672 19H5.03278Z"
+    fill="<%= color.brand.primary.value %>"/>
+</svg>
+```
+
+Notice the `<%= color.brand.primary.value %>` code in the SVG above. This is slightly different than how you would normally reference another design token in Style Dictionary. Inside the SVG we are using the [lodash template](https://lodash.com/docs/4.17.15#template) syntax for referencing data rather than the `{}` syntax Style Dictionary uses. I am processing this SVG code in a custom action, after all the tokens in the dictionary have been transformed and resolved. You could use another templating language like Handlebars if you wanted as well. The custom action code to output the finalized graphic assets would look something like this:
+
+```javascript
+const fs = require('fs-extra');
+const template = require('lodash/template');
+
+function(dictionary, config) {
+  const { androidPath, iosPath, buildPath } = config;
+  
+  dictionary.allProperties
+    .filter(token => {
+      return token.attributes.category === `image`
+    })
+    .forEach(token => {
+      const { name, value } = token;
+      
+      // Read the file from the token's value and turn it into a lodash template
+      const src = template( fs.readFileSync(value) );
+      
+      // Generate the final SVG output by passing in the dictionary
+      // to resolve the references.
+      const svg = src(dictionary.properties);
+      
+      // Make sure the directory exists and write the new SVG file
+      const outputPath = `${buildPath||''}${name}.svg`;
+      fs.ensureFileSync(outputPath);
+      fs.writeFileSync(outputPath, svg);
+      console.log(`✔︎  ${outputPath}`);
+
+      // Take the SVG and convert it into Android Vector Drawable format
+
+      // Take the SVG and convert it to a PNG and iOS imageset
+    });
+}
+```
+
+### Output references
+
+The second important part is the [`outputReferences`](https://amzn.github.io/style-dictionary/#/version_3?id=output-references) option that is coming out in [Style Dictionary 3.0](https://amzn.github.io/style-dictionary/#/version_3). Normally, Style Dictionary resolves all references (aka aliases) so that token values in output files are raw values. For example if you had these 2 tokens:
```

Comment:
For example if you had these 2 tokens: > For example, if you had these two tokens: